README.md

```diff
@@ -16,45 +80,114 @@ Generating this manifests is very easy, you just need add the proper dependency
+- The host under which the application is going to be exposed.
+- **`expose`** controls whether the application should be exposed via Ingress.
+- We use **`Always`** in order to be able to use an updated image.
 
-Generate the manifests launching a project compilation. Navigate to the directory and run `mvn clean package`. The generated manifests can be found under `target/classes/META-INF/dekorate.
+Generate the manifests launching a project compilation. Navigate to the directory and run `mvn clean package`. The generated manifests can be found under `target/classes/META-INF/dekorate`.
 
-NOTE: To perform the following steps you need to be connected to a running kubernetes cluster.
+## Building and deploying
 
-Run the following command to create the resources defined in the manifest.
+We need to somehow build and push the image to the registry to make it available for the kubernetes cluster. 
+
+### Using docker
+A basic Dockerfile is provided in the project base directory so you can build the image using the following command:
+```
+docker build -f Dockerfile -t USERNAME/hello-world-fwless:1.0-SNAPSHOT .
+```
+Then, tag the image as follows to be able to push it to the local registry:
+```
+docker tag USERNAME/hello-world-fwless:1.0-SNAPSHOT localhost:5000/hello-world-fwless:1.0-SNAPSHOT
 ```
-kubectl apply -f target/classes/META-INF/dekorate/kubernetes.yml
+Finally, you can push the image to the local registry
+```
+docker push localhost:5000/hello-world-fwless:1.0-SNAPSHOT
 ```
 
-At last, we need to expose the service in order to make it accessible from outside the cluster. For that, list the service using the command:
+As we have pushed the image to the local registry, we need to modify manually the kubernetes yml file generated to use that image.
+So, replace the following line: 
+```
+image: USERNAME/hello-world-fwless:1.0-SNAPSHOT
+```
+by
 ```
-oc get services
+image: localhost:5000/hello-world-fwless:1.0-SNAPSHOT
 ```
 
-Copy the name of the service and expose it running:
+Finally, we will deploy the application by posting the manifest with the following command:
 
 ```
-oc expose service hello-world-fwless
+kubectl create ns demo
+kubectl apply -f target/classes/META-INF/dekorate/kubernetes.yml -n demo
 ```
 
-Now, you should be able to get the address of the service and access the application with a browser. To get the adress, tape:
+Check the application by opening a browser to http://fw-app.127.0.0.1.nip.io/api/hello or curl http://fw-app.127.0.0.1.nip.io/api/hello
 
+Dekorate allows the user to trigger container image builds and deploy after the end of compilation, so the above actions can also be performed by executing the following command:
 ```
-kubectl get services
+mvn clean package -Ddekorate.build=true  -Ddekorate.push=true -Ddekorate.docker.registry="localhost:5000" -Ddekorate.deploy=true
 ```
 
-Open a browser to x.x.x.x:y.y/api/hello
+### Using jib-maven-plugin
 
```

Comment:
Add a few words here to explain why JIB and advantages vs dockerfile build used before ;-)